packages/react-i18n/src/babel-plugin/test/babel-plugin.test.ts

```diff
@@ -380,6 +380,122 @@ describe('babel-plugin-react-i18n', () => {
+                if ([].indexOf(locale) < 0) {
+                  return;
+                }
+
+                return (async () => {
+                  return {};
+                })();
+              }
+            });
+            return i18n.translate('key');
+          }
+        `,
+        ),
+      );
+    });
+
+    it('injects proper translation function for fallback plus one case', async () => {
+      expect(
+        await transformUsingI18nBabelPlugin(
+          useI18nFixture,
+          optionsForExplicitImportFile('MyComponent.tsx', 'fallbackPlusOne'),
+          {mode: 'with-explicit-paths'},
+        ),
+      ).toBe(
+        await normalize(
+          `import _en from './${TRANSLATION_DIRECTORY_NAME}/en.json';
+          import React from 'react';
+          import {useI18n} from '@shopify/react-i18n';
+
+          export default function MyComponent() {
+            const [i18n] = useI18n({
+              id: 'MyComponent_${defaultHash}',
+              fallback: _en,
+              translations(locale) {
+                if (["fr"].indexOf(locale) < 0) {
+                  return;
+                }
+
+                return (async () => {
+                  return await import(/* webpackChunkName: "MyComponent_${defaultHash}-i18n" */ "./${TRANSLATION_DIRECTORY_NAME}/fr.json").then((dict) => dict && dict.default);
+                })();
+              }
+            });
+            return i18n.translate('key');
+          }
+        `,
+        ),
+      );
+    });
+
+    it('injects translation function with multiple static imports', async () => {
+      expect(
+        await transformUsingI18nBabelPlugin(
+          useI18nFixture,
+          optionsForExplicitImportFile(
+            'MyComponent.tsx',
+            'multipleTranslations',
+          ),
+          {mode: 'with-explicit-paths'},
+        ),
+      ).toBe(
+        await normalize(
+          `import _en from './${TRANSLATION_DIRECTORY_NAME}/en.json';
+          import React from 'react';
+          import {useI18n} from '@shopify/react-i18n';
+
+          export default function MyComponent() {
+            const [i18n] = useI18n({
+              id: 'MyComponent_${defaultHash}',
+              fallback: _en,
+              translations(locale) {
+                if (["de", "fr", "zh-TW"].indexOf(locale) < 0) {
+                  return;
+                }
+
+                return (async () => {
+                  const returnDefault = (dict) => dict && dict.default;
+
+                  switch (locale) {
+                    case "de":
+                      return await import(/* webpackChunkName: "MyComponent_${defaultHash}-i18n" */ "./${TRANSLATION_DIRECTORY_NAME}/de.json").then(returnDefault);
+                    case "fr":
+                      return await import(/* webpackChunkName: "MyComponent_${defaultHash}-i18n" */ "./${TRANSLATION_DIRECTORY_NAME}/fr.json").then(returnDefault);
+                    default:
```

Comment:
Using default as the final option feels weird. Should this be `case: 'zh-TW'`?

Now that we're using `then` rather than await to deal with what to do with the result of a promise, I don't think we need the `await` keyword here anymore, `return import(...` might be enough. At that point, we're not using await, so I don't think we need this to be an immediately invoked async function anymore.

```js
 translations(locale) {
  if (["de", "fr", "zh-TW"].indexOf(locale) < 0) {
    return;
  }

  const returnDefault = (dict) => dict && dict.default;

   switch (locale) {
    case "de":
      return import(/* webpackChunkName: "MyComponent_${defaultHash}-i18n" */ "./${TRANSLATION_DIRECTORY_NAME}/de.json").then(returnDefault);
  // other cases here
  }
```